docs/user-guide/images.md

```diff
@@ -186,17 +186,50 @@ where node creation is automated.
 
 Kubernetes supports specifying registry keys on a pod.
 
-First, create a `.dockercfg`, such as running `docker login <registry.domain>`.
-Then put the resulting `.dockercfg` file into a [secret resource](secrets.md).  For example:
+First, login to your dockerhub registry by running `docker login` or to login to a private registry use `docker login <registry.domain>`.
+Docker will create a file named `config.json` under `~/.docker/`
 
 ```console
 $ docker login
 Username: janedoe
 Password: ●●●●●●●●●●●
 Email: jdoe@example.com
-WARNING: login credentials saved in /Users/jdoe/.dockercfg.
+WARNING: login credentials saved in /Users/jdoe/.docker/config.json
 Login Succeeded
+```
+
+This file has the following structure:
+
+```
+$ cat ~/.docker/config.json
+{
+	"auths": {
+		"https://index.docker.io/v1/": {
+			"auth": "ZmFrZXBhc3N3b3JkMTIK",
+			"email": "jdoe@example.com"
+		}
+	}
+}
+```
+
+Note that this is different than what is expected by the Kubernetes [secret resource](secrets.md): it expects a file named `.dockercfg` with the following structure:
+
+```console
+{ "https://index.docker.io/v1/": { "auth": "ZmFrZXBhc3N3b3JkMTIK", "email": "jdoe@example.com" } }
+```
+
+The notable differences are:
+- There is no `"auths"` wrapping
+- It must be on a single line.
+- The url must be prefixed with the url scheme `https://`: if you login to your private registry with `docker login <registry.domain>` without the `https://` scheme, it will not be present in the `config.json` but is required for the `.dockercfg` used by `imagePullSecret`.
+
+If you have multiple registries, they will show as multiple entries under `"auths":`. You will need to create an `imagePullSecret` for each entry.
+
+Copy the config.json file over to `~/.dockercfg` and modify to match the one-liner object above.
```

Comment:
Hrm - we support docker config.json as well as dockercfg natively - @deads2k I think this can be simplified to simply creating the secret with config.json and giving it the right type?